packages/select/index.js

```diff
@@ -36,29 +36,67 @@ export class WarpSelect extends kebabCaseAttributes(WarpElement) {
     // Whether to show optional text
     optional: { type: Boolean, reflect: true },
 
+    // Renders the field in a disabled state.
+    disabled: { type: Boolean, reflect: true },
+
+    // Renders the field in a readonly state.
+    readOnly: { type: Boolean, relfect: true },
+
     _options: { state: true },
+    _readOnlyId: { type: String },
   };
 
   static styles = [WarpElement.styles];
 
+  constructor() {
+    super();
+    activateI18n(enMessages, nbMessages, fiMessages);
+
+    this._options = this.innerHTML;
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    if (this.readOnly) {
+      this._readOnlyId = `w-select-readonly-${Math.random().toString(36).slice(2, 11)}`;
+      this.setAttribute('data-readonly-id', this._readOnlyId);
+      window.addEventListener('keydown', this.handleKeyDown);
+    }
+  }
+
+  disconnectedCallback() {
+    if (this.readOnly) window.removeEventListener('keydown', this.handleKeyDown);
+
+    super.disconnectedCallback();
+  }
+
+  handleKeyDown(event) {
+    const target = event.target.closest('w-select[data-readonly-id="' + this._readOnlyId + '"]');
+    if (target && (event.key === ' ' || event.key === 'ArrowDown' || event.key === 'ArrowUp')) {
+      event.preventDefault();
+    }
+  }
```

Comment:
Maybe we could utilise the `@keydown` event listener on the `select` element, like it was done in Card and Attention component. Then the `handleKeyDown" method could be more similar to what we have in React Select and the logic would be much less complicated :)

Reply:
That makes sense! I'll change it! 😄

Reply:
Updated in this commit: [0f706be](https://github.com/warp-ds/elements/pull/159/commits/0f706bee85360bd1ed77633b950f73f9f80f40d0)